torchrl/collectors/collectors.py

```diff
@@ -363,35 +371,56 @@ class SyncDataCollector(DataCollectorBase):
             If ``None`` is provided, the policy used will be a
             :class:`~torchrl.collectors.RandomPolicy` instance with the environment
             ``action_spec``.
+
+    Keyword Args:
         frames_per_batch (int): A keyword-only argument representing the total
             number of elements in a batch.
         total_frames (int): A keyword-only argument representing the total
             number of frames returned by the collector
             during its lifespan. If the ``total_frames`` is not divisible by
             ``frames_per_batch``, an exception is raised.
              Endless collectors can be created by passing ``total_frames=-1``.
-        device (int, str or torch.device, optional): The device on which the
-            policy will be placed.
-            If it differs from the input policy device, the
-            :meth:`~.update_policy_weights_` method should be queried
-            at appropriate times during the training loop to accommodate for
-            the lag between parameter configuration at various times.
-            Defaults to ``None`` (i.e. policy is kept on its original device).
+             Defaults to ``-1`` (endless collector).
+        device (int, str or torch.device, optional): The generic device of the
+            collector. The ``device`` args fills any non-specified device: if
+            ``device`` is not ``None`` and any of ``storing_device``, ``policy_device`` or
+            ``env_device`` is not specified, its value will be set to ``device``.
+            Defaults to ``None`` (No default device).
```

Comment:
Just to understand:

If my env is on cuda and i create a collector with `device=cpu` then my env is cast to cpu.
If, on the other hand, I want to preserve my env device, but have all other devices be cpu, I do `stroing_device=cpu, policy_device=cpu`

Reply:
the other option was that device would override all other devices but that's more messy and it makes all device kwargs useless as soon as you use device